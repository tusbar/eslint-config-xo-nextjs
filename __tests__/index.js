const lintFixture = require('../lib/lint-fixture')

describe('jsx-quotes', () => {
  it('correct', async () => {
    const result = await lintFixture('jsx-quotes.correct.js')

    expect(result.errorCount).toBe(0)
  })

  it('incorrect', async () => {
    const result = await lintFixture('jsx-quotes.incorrect.js')

    expect(result.messages).toHaveLength(1)
    expect(result.messages[0].ruleId).toBe('jsx-quotes')
    expect(result.messages[0].message).toBe('Unexpected usage of doublequote.')
  })
})

describe('react/jsx-closing-bracket-location', () => {
  it('correct', async () => {
    const result = await lintFixture('react/jsx-closing-bracket-location.correct.js')

    expect(result.errorCount).toBe(0)
  })
})

describe('react/jsx-no-duplicate-props', () => {
  it('correct', async () => {
    const result = await lintFixture('react/jsx-no-duplicate-props.correct.js')

    expect(result.errorCount).toBe(0)
  })

  it('incorrect', async () => {
    const result = await lintFixture('react/jsx-no-duplicate-props.incorrect.js')

    expect(result.messages).toHaveLength(1)
    expect(result.messages[0].ruleId).toBe('react/jsx-no-duplicate-props')
    expect(result.messages[0].message).toBe('No duplicate props allowed')
  })
})

describe('react/react-in-jsx-scope', () => {
  it('correct', async () => {
    const result = await lintFixture('react/react-in-jsx-scope.correct.js')

    expect(result.errorCount).toBe(0)
  })

  it('incorrect', async () => {
    const result = await lintFixture('react/react-in-jsx-scope.incorrect.js')

    expect(result.messages).toHaveLength(1)
    expect(result.messages[0].ruleId).toBe('no-unused-vars')
    expect(result.messages[0].message).toBe('\'React\' is defined but never used.')
  })
})
