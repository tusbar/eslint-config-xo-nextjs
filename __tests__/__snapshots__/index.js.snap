// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index should generate error messages 1`] = `
Array [
  Object {
    "column": 11,
    "endColumn": 32,
    "endLine": 1,
    "line": 1,
    "message": "'React' must be in scope when using JSX",
    "nodeType": "JSXOpeningElement",
    "ruleId": "react/react-in-jsx-scope",
    "severity": 2,
  },
]
`;

exports[`index should generate error messages 2`] = `
Array [
  Object {
    "column": 1,
    "endColumn": 6,
    "endLine": 1,
    "line": 1,
    "message": "'React' must be in scope when using JSX",
    "nodeType": "JSXOpeningElement",
    "ruleId": "react/react-in-jsx-scope",
    "severity": 2,
  },
  Object {
    "column": 2,
    "endColumn": 5,
    "endLine": 1,
    "line": 1,
    "message": "'App' is not defined.",
    "nodeType": "JSXIdentifier",
    "ruleId": "react/jsx-no-undef",
    "severity": 2,
  },
  Object {
    "column": 2,
    "endColumn": 11,
    "endLine": 2,
    "line": 2,
    "message": "'React' must be in scope when using JSX",
    "nodeType": "JSXOpeningElement",
    "ruleId": "react/react-in-jsx-scope",
    "severity": 2,
  },
  Object {
    "column": 3,
    "endColumn": 8,
    "endLine": 2,
    "line": 2,
    "message": "'Hello' is not defined.",
    "nodeType": "JSXIdentifier",
    "ruleId": "react/jsx-no-undef",
    "severity": 2,
  },
]
`;
